pkg/ipamd/ipamd.go

```diff
@@ -1197,20 +1215,74 @@ func (c *IPAMContext) logPoolStats(dataStoreStats *datastore.DataStoreStats) {
 	log.Debugf("%s: %s, c.maxIPsPerENI = %d", prefix, dataStoreStats, c.maxIPsPerENI)
 }
 
-func (c *IPAMContext) askForTrunkENIIfNeeded(ctx context.Context) {
-	if c.enablePodENI && c.dataStore.GetTrunkENI() == "" {
+func (c *IPAMContext) askForTrunkENIIfNeeded(ctx context.Context, labels map[string]string) error {
+	var err error
+	hasTrunkLabel := false
+
+	// if node labels is nil map which is rare, ipamd still sends the event and lets resource controller to handle this case
+	if labels != nil {
+		_, hasTrunkLabel = labels[trunkInterfaceLabel]
+	}
+
+	// check if trunk interface is not present OR trunk label is not present that covers special use case
+	// some special settings can boot up nodes with trunk interface attached already
+	// we still need to signal VPC RC to manage the nodes
+	if c.dataStore.GetTrunkENI() == "" || !hasTrunkLabel {
 		// Check that there is room for a trunk ENI to be attached:
 		if c.dataStore.GetENIs() >= (c.maxENI - c.unmanagedENI) {
-			log.Debug("No slot available for a trunk ENI to be attached. Not labeling the node")
-			return
+			log.Debug("No slot available for a trunk ENI to be attached. Will not create a node event for resource controller")
+			return nil
 		}
 		// We need to signal that VPC Resource Controller needs to attach a trunk ENI
-		err := c.SetNodeLabel(ctx, "vpc.amazonaws.com/has-trunk-attached", "false")
+		log.Debug("VPC CNI is asking RC to initialize trunk interface")
+		err = c.eventRecorder.SendNodeEvent(corev1.EventTypeNormal, eventrecorder.EventReason, sgpp.VpcCNINodeEventActionForTrunk, sgpp.TrunkEventNote)
 		if err != nil {
-			podENIErrInc("askForTrunkENIIfNeeded")
-			log.Errorf("Failed to set node label", err)
+			log.Error("Failed sending a node event to VPC RC for enabling trunk interface")
 		}
 	}
+
+	return err
+}
+
+func (c *IPAMContext) getNodeLabels(ctx context.Context) (map[string]string, error) {
+	var node corev1.Node
+	if err := c.cachedK8SClient.Get(ctx, types.NamespacedName{Name: c.myNodeName}, &node); err != nil {
+		return nil, err
+	} else {
+		return node.GetLabels(), err
+	}
+}
+
+func (c *IPAMContext) askForENIConfigIfNeeded(ctx context.Context, labels map[string]string) error {
+	var err error
+	hasLabelled := false
+
+	if err != nil {
```

Comment:
Why is this here? Missed during merge or cleanup?

Reply:
oh, this should have been removed after I combined the getNodeLabels into caller instead of making two calls in trunk and eniconfig methods. I will remove this. Thanks.